qiskit/providers/aer/extensions/multi_pauli.py

```diff
@@ -0,0 +1,75 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Simulator command to perform multiple pauli gates in a single pass
+"""
+import numpy as np
+
+from qiskit import QuantumCircuit, QuantumRegister
+from qiskit.circuit.library import XGate, YGate, ZGate
+from qiskit.circuit.gate import Gate
+
+
+class MultiPauliGate(Gate):
+    r"""A multi-qubit Pauli gate.
+
+        This gate exists for optimization purposes for the
+        quantum statevector simulation, since applying multiple
+        pauli gates to different qubits at once can be done via
+        a single pass on the statevector.
+
+        The functionality is equivalent to applying
+        the pauli gates sequentially using standard Qiskit gates
+        """
+
+    def __init__(self, pauli_string=None):
+        self.pauli_string = pauli_string
+        super().__init__('multi_pauli', len(pauli_string), [pauli_string[::-1]])
+
+    def _define(self):
+        """
+        gate multi_pauli (p1 a1,...,pn an) { p1 a1; ... ; pn an; }
+        """
+        gates = {'X': XGate, 'Y': YGate, 'Z': ZGate}
```

Comment:
This should support identity Pauli's too.

This could be done by changing the loop to:
```python
rules = [(gates[p](), [q[i]], []) for (i, p) in enumerate(self.pauli_string) if p != 'I']
```
or adding the `IGate` to the gates dict.

Reply:
I added IGate; it seems better than to omit the gate, since the user might want to put noise on the gate.